testutil/grpc.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"net"
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+// FakeGRPCServer creates and registers a fake grpc server for testing. It
+// returns the bound address and a connected client. It ensures the server is
+// stopped when tests finish. It returns an error if the connection does not
+// establish.
+func FakeGRPCServer(tb testing.TB, registerFunc func(*grpc.Server)) (string, *grpc.ClientConn) {
```

Comment:
Since we're pulling this into its own package with a public API, we should extract `register` func as a publicly-registered type.

Reply:
Why?

Reply:
It's better for documentation (godoc), so we can cleanly call out the invariants of the input function independent of the function definition itself (example http.HandlerFunc).

Reply:
It's not blocking for this (I approved the PR), but I think it would be nice for the godoc.

Reply:
Done